packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      expectNullFields(updateResult2.data[updateResultSetName], ['ownersContent']);
+
+      // user2 can update the private field
+      const todoUpdated3 = {
+        customId: todo['customId'],
+        privateContent: 'Private Content updated 1',
+      };
+      const updateResult3 = await user2TodoHelper.update(updateResultSetName, todoUpdated3, privateResultSet);
+      expect(updateResult3.data[updateResultSetName].customId).toEqual(todo['customId']);
+      expectNullFields(updateResult3.data[updateResultSetName], ['privateContent']);
+
+      // user2 can read the allowed fields
+      const getResult2 = await user2TodoHelper.get(
+        {
+          customId: todo['customId'],
+        },
+        privateResultSet,
+        false,
+        'all',
+        'customId',
+      );
+      checkOperationResult(
+        getResult2,
+        {
+          customId: todo['customId'],
+          privateContent: todoUpdated3['privateContent'],
+        },
+        `get${modelName}`,
+      );
+
+      const listTodosResult2 = await user2TodoHelper.list({}, privateResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['customId'], true, 'customId');
+
+      // owner(user1) can listen to updates on allowed fields
+      const todoRandom = {
+        ...todo,
+        customId: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        customId: todoRandom.customId,
+        authors: [userName1],
+        privateContent: 'Private Content updated',
+      };
+      const subscriberClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName1]);
+      const subTodoHelper = createModelOperationHelpers(subscriberClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await user1TodoHelper.create(createResultSetName, todoRandom, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, privateContent: null, ownersContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await user1TodoHelper.update(updateResultSetName, todoRandomUpdated, completeOwnerResultSet);
+          },
+        ],
+        {},
+        completeOwnerResultSet,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, privateContent: null, ownersContent: null },
+        `onUpdate${modelName}`,
+      );
+    });
```

Comment:
Should we also assert that user2 can update the `authors` field and delete the record? Or will the delete fail regardless because the `publicContent` field doesn't have owner delete permissions?

Reply:
For this model, delete fails, I have that case covered in custom owner field model where deletion is allowed.